### PR TITLE
plat-stm32mp1: bump to DTS files from Linux v5.6.10 and define flavors for ST boards.

### PR DESCRIPTION
**Pull request by etienne-lms**

### CODE REVIEW

**Comment by etienne-lms** [comment]
Checkpatch complains on header file **dt-bindings/input/linux-event-codes.h**, yet it is the header file copied unmodified from Linux kernel source file tree:
```
df96b207 dt-bindings: add stm32-pinfunc.h and generic input.h
CHECK: spaces preferred around that '+' (ctx:VxV)
#99: FILE: core/include/dt-bindings/input/linux-event-codes.h:51:
+#define EV_CNT			(EV_MAX+1)
               			       ^

CHECK: spaces preferred around that '+' (ctx:VxV)
#110: FILE: core/include/dt-bindings/input/linux-event-codes.h:62:
+#define SYN_CNT			(SYN_MAX+1)
                			        ^

WARNING: line over 80 characters
#280: FILE: core/include/dt-bindings/input/linux-event-codes.h:232:
+#define KEY_ROTATE_DISPLAY	153	/* Display orientation for e.g. tablets */

...
```

File: core/arch/arm/plat-stm32mp1/conf.mk [hunk 1]
```diff
@@ -1,9 +1,33 @@
-PLATFORM_FLAVOR ?= stm32mp157
```
Comment (jforissier) [inline core/arch/arm/plat-stm32mp1/conf.mk:1]: Don't you want to keep a default value? What will get built by `make PLATFORM=stm32mp1`?

File: core/include/dt-bindings/input/input.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/* SPDX-License-Identifier: GPL-2.0 */
```
Comment (jforissier) [inline core/include/dt-bindings/input/input.h:1]: We may have a problem with this (no BSD-compatible license)

File: core/include/dt-bindings/input/linux-event-codes.h [hunk 3]
```diff
@@ -0,0 +1,864 @@
+/* SPDX-License-Identifier: GPL-2.0 WITH Linux-syscall-note */
```
Comment (jforissier) [inline core/include/dt-bindings/input/linux-event-codes.h:1]: We may have a problem with this (no BSD-compatible license)

File: core/include/dt-bindings/input/input.h (hunk 2 above)
Comment (etienne-lms) [reply core/include/dt-bindings/input/input.h:1]: Oh! I missed this one.
Until this get solved in a nice way and as stm32mp1 OP-TEE Core does not deal with input bindings, I think I'll modify the DTS file remove data related to these bindings.
Sound fair?

File: core/include/dt-bindings/input/linux-event-codes.h (hunk 3 above)
Comment (etienne-lms) [reply core/include/dt-bindings/input/linux-event-codes.h:1]: ditto.

File: core/arch/arm/plat-stm32mp1/conf.mk (hunk 1 above)
Comment (etienne-lms) [reply core/arch/arm/plat-stm32mp1/conf.mk:1]: `make PLATFORM=stm32mp1` should build the base platform without any embedded DTB as prior this change.

**Comment by etienne-lms** [comment]
License issue addressed.
Comments on commit messages not yet addressed.

File: core/include/dt-bindings/input/input.h (hunk 2 above)
Comment (jforissier) [reply core/include/dt-bindings/input/input.h]: Fine with me. Perhaps you could further minimize the changes by adding the 5 needed defines at the beginning of the file instead? (`KEY_ENTER` etc.). As you like.

**Comment by jforissier** [comment]
A general comment: there are quite a lot of things in those DTS files, is everything needed in the OP-TEE repository (and in the resulting binary `.dtb`)? Or perhaps some parts could be filtered out?

Anyway:
`Acked-by: Jerome Forissier <jerome@forissier.org>`

File: core/include/dt-bindings/input/input.h (hunk 2 above)
Comment (etienne-lms) [reply core/include/dt-bindings/input/input.h]: I guess that would not clearly tackle the license issue.
I would prefer to wait a cleaner way. The system device tree initiative (sharing DTS files among firmwares in a system) should address this in one way or another I guess.

**Comment by etienne-lms** [comment]
> A general comment: there are quite a lot of things in those DTS files, is everything needed in the OP-TEE repository (and in the resulting binary .dtb)? Or perhaps some parts could be filtered out?

You are right. A lot of the embedded content could be filtered out. This needs specific tools I don't want to address here. I prefer to wait for a generic solution (system device tree?). Until then, I prefer to propose a easy way for one to share DTS files between Linux kernel and OP-TEE hence restrict changes to the minima when syncing DTS files between the 2 components.

I'll squash the fixup commits.

**Comment by etienne-lms** [comment]
I squashed the fixup commits, modified commit messages and applied the review tag.

**Pull request merged**